### DISCUSSION TITLE
Breaking change when going from 17.0.3 to 17.10.2

### DISCUSSION THREAD

**Discussion by egil** [VerifyTests/Verify#612] [Q&A]
Hi @SimonCropp

I was surprised when my tests started failing after upgrading  from 17.0.3 to 17.10.2. The issue seems to two fold:

First, the framework version is now being added to "received files", here is an example with before an after:

```diff
- RetrieveResourceEndpointTest.RetrieveById_with_IfNoneMatch_header_returns_200_when_modified.received.txt
+ RetrieveResourceEndpointTest.RetrieveById_with_IfNoneMatch_header_returns_200_when_modified.DotNet6_0.received.txt
```

Second, the GUID scrubbing has changed, or ordering of numbers assigned have. Here is the diff:

```diff
{
  Version: 1.1,
  Status: 200 OK,
  Headers: {
-   ETag: "Guid_2"
+   ETag: "Guid_1"
  },
  Cookies: {},
  Content: {
    Headers: {
-     Content-Location: http://localhost/v2/Users/Guid_1,
+     Content-Location: http://localhost/v2/Users/Guid_2,
      Content-Type: application/scim+json
    },
    Value: {
-    id: Guid_1,
+    id: Guid_2,
      meta: {
        created: DateTime_1,
        lastModified: DateTime_1,
-       location: http://localhost/v2/Users/Guid_1,
+       location: http://localhost/v2/Users/Guid_2,
        resourceType: User,
-       version: "Guid_2"
+       version: "Guid_1"
      },
      schemas: [
        urn:ietf:params:scim:schemas:core:2.0:User
      ],
      userName: userName_Guid_3
    }
  },
  Request: {
-   Uri: http://localhost/v2/Users/Guid_1,
+   Uri: http://localhost/v2/Users/Guid_2,
    Headers: {
      If-None-Match: "Guid_4"
    },
    Cookies: {}
  }
}
```

The second issue I can probably get around by just accepting all changes automatically. The first I need help figuring out how to configure my way out of. I don't want the framework version in the received/verified files. My test framework only targets .NET 6.

**Comment by SimonCropp** [comment]
the change in name of received files should not have broken any tests. if it did can u submit a pr with the scenario that caused it

i will be looking into detection if a project is targeting multiple frameworks, and using that to decide if the framework name+version goes in the received file

the guid scrubbing seem to be more correct looking at the diff. is u think it is wrong can u submit a failing pr

note that i think these changes are in keeping with the documented approach to versioning https://github.com/VerifyTests/Verify#versioning

**Reply by egil** [reply]
Hmm fair enough, had not read the notes on semver in verify, so was not expecting my tests go go red after a minor upgrade.

My verified/received files are already at the edge of how long filenames can be on Windows, so would definitely prefer not to have unneeded content in the filenames.

Is there a way to disable framework being included in the current release or is it something you need to enable?

**Reply by SimonCropp** [reply]
assuming you only target one framework, can u try 18.0.0-beta.8

**Reply by SimonCropp** [reply]
@egil did the new beta work for you?

**Reply by egil** [reply]
Sorry, had a busy end of the week. I'll try out the beta on Monday.

**Reply by egil** [reply]
It seems to take care of the net6 being added to file names. However, it does change the order a bit on the content of json.

**Reply by SimonCropp** [reply]
so this is how it should be working https://github.com/VerifyTests/Verify/issues/619

if that is not the case can u share a repro?
